content/client-lib-development-guide/features.textile

```diff
@@ -1473,6 +1495,82 @@ enum StatsIntervalGranularity:
   DAY
   MONTH
 
+class DeviceDetails:
+  + fromLocalDevice() => io DeviceDetails
+  id: String
+  platform: DevicePlatform
+  formFactor: DeviceFormFactor
+  clientId: String?
+  metadata: Dict<String, String>
+  updateTokenId: String
+  resetId(String)
+  resetUpdateToken() => io
+  push: DevicePushDetails
+
+class DevicePushDetails:
+  transportType: DevicePushTransportType
+  transportId: String
+  transportToken: String?
+  state: .Active | .Failing | .Failed
+  errorReason: ErrorInfo?
+  metadata: Dict<String, String>
+
+class Push:
+  publish(options: Dict<String, String>, JsonObject) => io
```

Comment:
is options not params?  params used below for get  / remove etc. and these options are required to publish so I would argue options is the wrong word to describe the argument.